flake.nix

```diff
@@ -0,0 +1,47 @@
+{
+  description = "Anarchy AI - A highly optimized and opinionated backend for running LLMs";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    nixgl.url = "github:guibou/nixGL";
```

Comment:
https://github.com/numtide/nix-gl-host/ is often a much simpler solution for FHS distributions (it just uses driver libraries from the host)

Reply:
My assumption was that `nixglDefault` would try to find the drivers and use them if available. From what I can see on both READMEs they're trying to accomplish the same thing, although I don't know too much about nix-gl-host.

Reply:
- nixglhost uses whatever drivers deployed by the host system (e.g. if you run it on ubuntu, it'll find and use the drivers deployed by ubuntu); sometimes using the host system's libraries is the only choice (e.g. nvidia jetsons).
- nixGL tries to choose a nixpkgs attribute which would provide the equivalent drivers but packaged by nix (and patchelfed to use the same libc &c as your project).

There are pros and cons for both, they strike different tradeoffs.

Reply:
Also,

> (the impure is needed here in order for [nixGL](https://github.com/nix-community/nixGL) to find the host nvidia drivers)

I think you could avoid `--impure`. I think the only reason it might need impure is for `<nixpkgs>`. The nixpkgs revision I think you want to use for the drivers is the revision compatible with your project, i.e. the same nixpkgs you pin in the `flake.lock`. So it should be ok to just pass it